### DISCUSSION TITLE
Add much data to zset and flushall, memory not free.

### DISCUSSION THREAD

**Discussion by cwjioai** [microsoft/garnet#871] [Q&A]
Operator System

Windows 11 24h2

Config file

https://github.com/microsoft/garnet/blob/main/libs/host/defaults.conf

Server Command
```
GarnetServer.exe --config-import-path garnet.conf
```

Client Command
```
redis-benchmark -n 10000000 -r 10000000 -q -P 16 zadd zset __rand_int__ __rand_int__

redis-cli
> keys *
1) "zset"
> flushall
OK
> keys *
(empty array)
```

Result

When I executed the benchmark command about 5 times, the memory usage increased to 6GB. Executing flushall did not reduce the memory, and trying forcegc was also ineffective.

I also modify the memory config in file, and not work.

![Snipaste_2024-12-11_13-35-29](https://github.com/user-attachments/assets/a557e918-31a6-4d48-90f4-3dfda0fdf9c1)

**Comment by badrishc** [comment]
Try disabling server GC in the main csproj.
